tests/phpunit/AutoReview/IntegrationGroup/IntegrationGroupProvider.php

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /**
+     * @return string[]|null
+     */
+    private static function checkIoOperations(string $className): ?array
+    {
+        $classReflection = new ReflectionClass($className);
+
+        $testCaseClass = SourceTestClassNameScheme::getTestClassName($className);
+
+        try {
+            $testCaseReflection = new ReflectionClass($testCaseClass);
+        } catch (ReflectionException $exception) {
+            // No test case could be find for this source file
+            return null;
+        }
+
+        //
+        // Check the test cases code
+        //
+        $testCaseFileName = $testCaseReflection->getFileName();
+        $testCaseCode = file_get_contents($testCaseFileName);
+
+        // Case where the test case itself use I/O functions
+        if (self::codeContainsIoFunctions($testCaseCode)) {
+            return [$testCaseClass, $testCaseFileName];
+        }
+
+        $parentTestCaseClassReflection = $testCaseReflection->getParentClass();
+
+        Assert::isInstanceOf($parentTestCaseClassReflection, ReflectionClass::class);
+
+        while ($parentTestCaseClassReflection->getName() !== TestCase::class) {
+            if ($parentTestCaseClassReflection->getName() === FileSystemTestCase::class) {
+                return [$testCaseClass, $parentTestCaseClassReflection->getFileName()];
+            }
+
+            $parentTestCaseFileName = $parentTestCaseClassReflection->getFileName();
+
+            $testCaseCode = file_get_contents($parentTestCaseFileName);
+
+            if (self::codeContainsIoFunctions($testCaseCode)) {
+                return [$testCaseClass, $parentTestCaseFileName];
+            }
+
+            $parentTestCaseClassReflection = $parentTestCaseClassReflection->getParentClass();
+
+            Assert::isInstanceOf($parentTestCaseClassReflection, ReflectionClass::class);
+        }
+
+        //
+        // Check the source class code
+        //
+        $classFileName = $classReflection->getFileName();
+        $classCode = file_get_contents($classFileName);
+
+        // Case where the test case itself use I/O functions
+        if (self::codeContainsIoFunctions($classCode)) {
+            return [$testCaseClass, $classFileName];
+        }
+
+        $parentClassReflection = $classReflection->getParentClass();
+
+        while ($parentClassReflection !== false) {
+            $parentClassFileName = $parentClassReflection->getFileName();
+
+            if ($parentClassFileName === false) {
+                break;
+            }
+
+            $parentClassCode = file_get_contents($parentClassFileName);
+
+            if (self::codeContainsIoFunctions($parentClassCode)) {
+                return [$testCaseClass, $parentClassFileName];
+            }
+
+            $parentClassReflection = $parentClassReflection->getParentClass();
+        }
+
+        return null;
+    }
+
+    private static function codeContainsIoFunctions(string $code): bool
```

Comment:
what about `Filesystem::**` checks? If that's hard to implement, let's ignore that for now ;)